### ISSUE TITLE
First‐order data generator

### ISSUE THREAD

**Issue by Cortys** [Cortys/SEP-Probly-WS2526#21]
Recently, [_credal sets_](https://arxiv.org/abs/2505.22332) have gained traction as a suitable way to represent uncertainty. One important metric to evaluate such sets is _coverage_, which says whether the ground-truth conditional distribution is contained in the set or not. To compute this metric, ground-truth conditional distribution, which may also be referred to as **first-order data**, is required. This epic aims to implement an approach to generate such data. 

### Background
In machine learning, we usually assume that the data follows some (joint) distribution $p(\boldsymbol{X},Y)$ where $\boldsymbol{X}$ and $Y$ are random variables representing the input features and targets, respectively, which gives rise to the conditional distribution $p(Y|\boldsymbol{X}) = \frac{p(Y|\boldsymbol{X})}{p(\boldsymbol{X})}$. Normally, we don't have access to these distributions, hence we need to resort to approximations. One way to approximate the conditional distributions is to use the samples $\boldsymbol{x}$ as input to a "good" pre-trained model from e.g. Huggingface (denote as $\hat{h}$), thereby creating approximations $\hat{h}(\boldsymbol{x}) \approx p(\cdot \mid \boldsymbol{x})$. This epic aims to implement this as an easy-to-use feature in `probly`. 

### Goals
- Implement a class or function (discuss with product owner / stakeholders) that takes in any model and an existing dataset and outputs a dataset or DataLoader that can be for training and evaluation of models. 
- Implement an easy and efficient way for users to save the ground-truth distributions that they generate. Obviously, we only want to save the distributions, and not the outputs, but there needs to be a good system in place to match them after loading the distributions.

### Constraints 
None so far.
